src/calories.rs

```diff
@@ -0,0 +1,92 @@
+#[derive(Debug, PartialEq)]
+pub struct CalorieCounter {
+    third_max: usize,
+    second_max: usize,
+    pub max: usize,
+    current_counter: usize,
+}
+
+impl CalorieCounter {
+    pub fn new(
+        third_max: usize,
+        second_max: usize,
+        max: usize,
+        current_counter: usize,
+    ) -> CalorieCounter {
+        CalorieCounter {
+            third_max,
```

Comment:
Having explicit third/second/first variables feels a bit nasty.  Feels more intuitive perhaps to use an array here `max_calories: usize[3]`  - also makes the code easier to generalize to larger values of N?  Though then that does make the struct update plan below less nice.